nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,73 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    console.log(meals);
+    console.log("in /api/meals");
+    let selectedMeals = meals;
+
+    //to filter using maxPrice
+    if ('maxPrice' in request.query) {
+      const maxPrice = parseInt(request.query.maxPrice);
+      if (isNaN(maxPrice)) {
+        res.status(400).send({ error: "maxPrice must be an integer" });
+        return;
+      }
+      selectedMeals = selectedMeals.filter(meal => meal.price < maxPrice);
+    }
+
+    //to filter using title
+    if ('title' in request.query) {
+      const title = request.query.title;
+      selectedMeals = selectedMeals.filter(meal => meal.title.toLowerCase().includes(title));
+    }
+    
+    //to filter using the date created
+    if ('createdAfter' in request.query) {
+      const createdAfter = new Date(request.query.createdAfter);
+      if (!createdAfter.getDate()) {
+        response.status(404).send('must be a valid date.');
+        return;
+      }
+      selectedMeals = selectedMeals.filter(meal => new Date(meal.createdAt) > createdAfter);
+    }
+
+    //to filter using the limit of meals
+    if ('limit' in request.query) {
+      const limit = parseInt(request.query.limit);
+      if (isNaN(limit)) {
+        res.status(400).send({ error: "limit must be an integer" });
+        return;
+      }
+      selectedMeals = selectedMeals.slice(0, limit);
+    }
+    response.send(selectedMeals);
+  }
+  catch (error) {
+    throw error;
+  }
+});
+
+//to filter meals by using the given id
+router.get("/:id", async (request, response) => {
+  try {
+    const id = parseInt(request.params.id);
+    if (isNaN(id)) {
+      response.status(404).send('User IDs must be integers.');
```

Comment:
.... I think 400 (bad request) is better than 404. here you can find different return codes that you can use https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
```suggestion
      response.status(400).send('User IDs must be integers.');
```